activation/poetdb.go

```diff
@@ -34,18 +34,27 @@ func NewPoetDb(db *sql.Database, log log.Log) *PoetDb {
 
 // HasProof returns true if the database contains a proof with the given reference, or false otherwise.
 func (db *PoetDb) HasProof(proofRef []byte) bool {
-	_, err := db.GetProofMessage(proofRef)
-	return err == nil
+	has, err := poets.Has(db.sqlDB, proofRef)
+	return err == nil && has
 }
 
 // ValidateAndStore validates and stores a new PoET proof.
 func (db *PoetDb) ValidateAndStore(proofMessage *types.PoetProofMessage) error {
+	ref, err := proofMessage.Ref()
+	if err != nil {
+		return err
```

Comment:
@countvonzero do we skip `fmt.Errorf` here on purpose?

Reply:
more like a Freudian slip on my part. 
not a fan of suing fmt.Errorf, especially when the code stack is deep. just make the error unreadable.
prefer to use context for debugging instead.

Reply:
Interesting point :) On the contrary I try to wrap as much as possible everywhere as then you see the whole execution stack in one line and don't need anything else to find the issue :)